src/ui/searchdialog.cc

```diff
@@ -161,37 +161,71 @@ qint64 SearchDialog::findNext(bool include_overlapping, bool interactive) {
     }
   }
 
-  if (backwards) {
-    start_search_pos_modifier = -(start_search_pos_modifier - 1);
-  }
-
   qint64 start_search_pos = _lastFoundPos + start_search_pos_modifier;
 
   qint64 idx = -1;
-  if (backwards) {
-    idx = lastIndexOf(_findBa, start_search_pos);
-  } else {
-    idx = indexOf(_findBa, start_search_pos);
-  }
+  idx = indexOf(_findBa, start_search_pos);
 
   if (idx >= 0) {
     _hexEdit->setSelection(idx, _findBa.size(), true);
     _lastFoundSize = _findBa.size();
     emit enableFindNext(true);
   } else {
     _lastFoundSize = 0;
-    if (backwards) {
-      _hexEdit->setSelection(_hexEdit->dataModel()->binData().size() - 1, 0,
-                             false);
+    _hexEdit->setSelection(0, 0, false);
+
+    if (interactive) {
+      message_box_not_found_->show();
+    }
+  }
+
+  return idx;
+}
+
+qint64 SearchDialog::findPrevious(bool include_overlapping, bool interactive) {
+  emit enableFindPrevious(false);
+
+  _findBa =
+      getContent(ui->cbFindFormat->currentIndex(), ui->cbFind->currentText());
+
+  if (_findBa.size() == 0) {
+    return -1;
+  }
+
+  bool overlapping = ui->cbOverlapping->isChecked() && include_overlapping;
+
+  qint64 start_search_pos_modifier = 0;
+  if (_lastFoundSize > 0) {
+    if (overlapping) {
+      start_search_pos_modifier = 1;
     } else {
-      _hexEdit->setSelection(0, 0, false);
+      start_search_pos_modifier = _lastFoundSize;
     }
+  }
+
+  start_search_pos_modifier = -(start_search_pos_modifier - 1);
+
+  qint64 start_search_pos = _lastFoundPos + start_search_pos_modifier;
+
+  qint64 idx = -1;
+  idx = lastIndexOf(_findBa, start_search_pos);
+
+  if (idx >= 0) {
+    _hexEdit->setSelection(idx, _findBa.size(), true);
+    _lastFoundSize = _findBa.size();
+    emit enableFindPrevious(false);
```

Comment:
I think it should be `true` instead of `false`. This signal is not used so it won't change anything, but this may change in near future when we add more buttons to the hexedit toolbar (`enableFindNext` is connected to the "Find next" button in hex edit).

Reply:
Fixed